### PR TITLE
Factorize tasks

### PR DESCRIPTION
**Pull request by delhomer**
This PR aims at factorizing `bordeaux.py` and `lyon.py` into a unified `city.py`, and as a consequence, to make the addition of a new city easier.

The whole Luigi pipeline has been reviewed in this view, from station data dowloading to machine learning algorithm execution.

Some improvements are still possible, especially regarding the downloading URLs handling. I suggest to open a further PR to address such points, if necessary.

This PR fixes issue #5 , by definition, and #14 as database table features as well as `csv` intermediary file features have been normalized.

### CODE REVIEW

**Comment by garaud** [comment]
Two important tasks to do after that:
* update the tables in our db to have the same names/columns
* update the SQL queries from the controller.py since the name of the tables and columns will be modified

File: jitenshea/tasks/city.py [hunk 1]
```diff
@@ -387,25 +469,29 @@ def run(self):
...
-                                          tablename=config['bordeaux']['clustering'])
+
     columns = [('station_id', 'INT'),
                ('start', 'DATE'),
                ('stop', 'DATE'),
                ('cluster_id', 'INT')]
 
+    @property
+    def table(self):
+        return '{schema}.clustered_stations'.format(schema=self.city)
```
Comment (garaud) [inline jitenshea/tasks/city.py]: the name of the cluster table should come from the configuration file.

File: jitenshea/tasks/city.py [hunk 2]
```diff
@@ -1,35 +1,36 @@
...
 from datetime import datetime as dt
 from datetime import date, timedelta
 
 from lxml import etree
 
-import sh
-
-import requests
-
 import numpy as np
```
[the remark this answers is not in the document]
Comment (delhomer) [reply jitenshea/tasks/city.py]: You're write, it comes from a wrong copy-paste process.

File: jitenshea/tasks/city.py [hunk 3]
```diff
@@ -1,35 +1,36 @@
...
+datasets: https://gitlab.com/Oslandia/tempus_demos
 
-Note: the 'ident' field which should be used for an unique id for each station
-is different when you load the layer TB_STVEL_P and CI_VCUB_P.
+Supported cities:
 
-  - TB_STVEL_P: bicycle-station geoloc
-  - CI_VCUB_P: bicycle-station real-time occupation data
+* Bordeaux
+  - stations URL:
```
[the remark this answers is not in the document]
Comment (delhomer) [reply jitenshea/tasks/city.py]: Yes, absolutely. I wrote the docstring before to find the good URLs, and did not go back to it when I finally got them.

File: jitenshea/tasks/city.py (hunk 1 above)
Comment (garaud) [reply jitenshea/tasks/city.py]: You're right. We shouldn't have a tablename for each city. But I like to have the possibility to choose these table names. We could move the name to the `database` section.

**Comment by garaud** [comment]
The parameters were set in this commit https://github.com/garaud/jitenshea/commit/bb5878912ec1d784df34e29d4f193e81be14200a

**Comment by garaud** [comment]
Note I change the way to normalize the table. The query is just more robust, specially about the projection of the Geometry type.

See https://github.com/garaud/jitenshea/pull/24/commits/711c782de88206ef33eb6b5ac7175875fb43e622

**Comment by delhomer** [comment]
That's perfect for me, except the fact that in my humble opinion, we should generalize this choice to every implied table (`raw_stations`, `stations` and `timeseries` were not yet considered). I've just pushed a commit to fix that.

From now, if everything is OK with this last commit, we can merge the PR.

**Pull request merged**